### PR TITLE
Update books page (add Horstmann, update FPiS, reshuffle)

### PR DESCRIPTION
**Pull request by SethTisue**
updates https://docs.scala-lang.org/books.html

the Horstmann book used to be there, but was removed (in #1706) because it had fallen out of date, but there's a Scala 3 edition now so let's bring it back

### CODE REVIEW

**Comment by SethTisue** [comment]
@mpilquist you might want to check this

**Comment by SethTisue** [comment]
CI failures is a lightbend.com 404 unrelated to my changes (and I'm taking it up with Lightbend IT)

File: _books/7-functional-programming-in-scala.md [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-authors: ["Paul Chiusano", "Rúnar Bjarnason"]
+status: Updated for Scala 3
+authors: ["Michael Pilquist", "Paul Chiusano", "Rúnar Bjarnason"]
 publisher: Manning
 publisherLink: https://www.manning.com/
 ---
 
 "Functional programming (FP) is a style of software development emphasizing functions that don't depend on program state... Functional Programming in Scala is a serious tutorial for programmers looking to learn FP and apply it to the everyday business of coding. The book guides readers from basic techniques to advanced topics in a logical, concise, and clear progression. In it, you'll find concrete examples and exercises that open up the world of functional programming."
 
 Foreword by Martin Odersky.
```
Comment (mpilquist) [inline _books/7-functional-programming-in-scala.md:13]: I suggest "Forewords by Daniel Spiewak and Martin Odersky."

File: _books/7-functional-programming-in-scala.md [hunk 2]
```diff
@@ -1,15 +1,13 @@
 ---
 title: "Functional Programming in Scala"
-link: https://www.manning.com/books/functional-programming-in-scala
+link: https://www.manning.com/books/functional-programming-in-scala-second-edition
 image: /resources/img/books/FPiS_93x116.png
```
Comment (mpilquist) [inline _books/7-functional-programming-in-scala.md:4]: Suggest updating the image to the new cover.

**Comment by mpilquist** [comment]
Thanks @SethTisue - added a couple small comments.

**Review by bishabosha — APPROVED** [review]

**Pull request merged**